mapo/agents/mapo/losses.py

```diff
@@ -0,0 +1,130 @@
+"""Collection of loss functions in tensorflow."""
+import tensorflow as tf
+from tensorflow import keras
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.evaluation.postprocessing import Postprocessing
+
+from mapo.kernels import KERNELS
+
+
+def dynamics_mle_loss(batch_tensors, model):
+    """Compute dynamics loss via Maximum Likelihood Estimation."""
+    obs, actions, next_obs = (
+        batch_tensors[SampleBatch.CUR_OBS],
+        batch_tensors[SampleBatch.ACTIONS],
+        batch_tensors[SampleBatch.NEXT_OBS],
+    )
+    return -tf.reduce_mean(model.compute_states_log_prob(obs, actions, next_obs))
+
+
+def dynamics_pga_loss(batch_tensors, model, actor_loss, config):
+    """Compute Policy Gradient Aware dynamics loss"""
+    gmapo = tf.gradients(actor_loss, model.actor_variables)
+    dpg_loss = actor_dpg_loss(batch_tensors, model)
+    dpg = tf.gradients(dpg_loss, model.actor_variables)
+    kernel = KERNELS[config["kernel"]]
```

Comment:
As you can see I passed the kernel as a string (l2, l1, cos_similarity, ...)... I didn't see the use of passing the lambda itself with ray.tune.function... if this is not fine, just let me know.

Reply:
Nice! I agree it's better now.